### PR TITLE
Cardgame gold sol

### PR DESCRIPTION
**Pull request by HarryWangATX**

### CODE REVIEW

File: solutions/USACO dec15-high-card-low-card-gold.mdx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+			}
+		}
+
+		ix = 0;
+		for (int i = 0; i < S.size(); i++) {
+			if(ix < B.size() && B.get(ix) < S.get(i)){
+				ix++;
+				score++;
+			}
+		}
```
Comment (frodakcin) [inline solutions/USACO dec15-high-card-low-card-gold.mdx:66]: I'm fairly certain `ix >= 0`, `B.remove(ix);`, and `ix < B.size()` are unnecessary. Whether or not you should remove them is up to you.

File: solutions/USACO dec15-high-card-low-card-gold.mdx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	static int[] C;
+	static List<Integer> B = new ArrayList<>();
+	static List<Integer> F = new ArrayList<>();
+	static List<Integer> S = new ArrayList<>();
+	static int N;
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio("cardgame");
+		N = io.nextInt();
+
+		C = new int[2*N+1];
```
Comment (frodakcin) [inline solutions/USACO dec15-high-card-low-card-gold.mdx:31]: Can be boolean.

File: solutions/USACO dec15-high-card-low-card-gold.mdx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+			else
+				S.add(c);
+		}
+
+		for (int i = 1; i < 2 * N + 1; i++) {
+			if(C[i] == 0) B.add(i);
+		}
+
+		Collections.sort(F);
+		Collections.sort(B);
```
Comment (frodakcin) [inline solutions/USACO dec15-high-card-low-card-gold.mdx:47]: Technically redundant (B is already sorted).

**Review by frodakcin — APPROVED** [review]
Optional changes. LMK if you want to change or ignore them, otherwise it's ready to merge I think

File: solutions/USACO dec15-high-card-low-card-gold.mdx (hunk 1 above)
Comment (HarryWangATX) [reply solutions/USACO dec15-high-card-low-card-gold.mdx:66]: true did not notice!!

File: solutions/USACO dec15-high-card-low-card-gold.mdx (hunk 3 above)
Comment (HarryWangATX) [reply solutions/USACO dec15-high-card-low-card-gold.mdx:47]: true

File: solutions/USACO dec15-high-card-low-card-gold.mdx (hunk 2 above)
Comment (HarryWangATX) [reply solutions/USACO dec15-high-card-low-card-gold.mdx:31]: sure

**Comment by HarryWangATX** [comment]
@frodakcin i fixed

File: solutions/USACO dec15-high-card-low-card-gold.mdx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+			if(i < N/2)
+				F.add(c);
+			else
+				S.add(c);
+		}
+
+		for (int i = 1; i < 2 * N + 1; i++) {
+			if(!C[i]) B.add(i);
+		}
+
```
Comment (frodakcin) [inline solutions/USACO dec15-high-card-low-card-gold.mdx:45]:
```suggestion

		// Sort cards (B is already sorted)
```

**Review by frodakcin — CHANGES_REQUESTED** [review]
Maybe add comments explaining what B, F, and S are.

**Review by frodakcin — APPROVED** [review]
ok

**Comment by frodakcin** [comment]
I'll merge in 2 hours if nothing else happens

**Pull request merged**